.github/workflows/benchmark.yml

```diff
@@ -31,15 +31,24 @@ jobs:
         uses: actions/setup-python@v4.5.0
         with:
           python-version: "3.11"
+      - name: Cache rust and pip
+        uses: ./.github/actions/cache
+        timeout-minutes: 2
+        with:
+          key: benchmark-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('cryptography-main/**/Cargo.lock') }}
 
       - name: Create virtualenv (main)
         run: |
           python -m venv .venv-main
           .venv-main/bin/pip install -v -c ./cryptography-main/ci-constraints-requirements.txt "./cryptography-main[test]" ./cryptography-main/vectors/
+        env:
+          CARGO_TARGET_DIR: ${{ format('{0}/src/rust/target/', github.workspace) }}
       - name: Create virtualenv (PR)
         run: |
           python -m venv .venv-pr
           .venv-pr/bin/pip install -v -c ./cryptography-pr/ci-constraints-requirements.txt "./cryptography-pr[test]" ./cryptography-pr/vectors/
+        env:
+          CARGO_TARGET_DIR: ${{ format('{0}/src/rust/target/', github.workspace) }}
```

Comment:
So this will share the target/ dir between the main and PR, do we want that?

I intentionally did not include any caching in this job because I wanted some assurance of no bleed between jobs.

Reply:
Is there any scenario where that would be unsafe? Any changed rust files will have differing intermediate artifact names right?

That said, we could just drop it from the PR and have no caching for that part.

Reply:
It _shouldn't_ have an impact, but I think I have a soft-preference for not including it out of an abundance of cautioun -- and also since this is far from the slowest job.